Common/Securities/Future/FutureFilterUniverse.cs

```diff
@@ -19,40 +19,16 @@
 using QuantConnect.Data;
 using System.Linq;
 using System.Collections;
+using QuantConnect.Securities.Future;
 using QuantConnect.Util;
 
 namespace QuantConnect.Securities
 {
     /// <summary>
     /// Represents futures symbols universe used in filtering.
     /// </summary>
-    public class FutureFilterUniverse : IDerivativeSecurityFilterUniverse
+    public class FutureFilterUniverse : ContractSecurityFilterUniverse<FutureFilterUniverse>
     {
-        internal IEnumerable<Symbol> _allSymbols;
-
-        /// <summary>
-        /// The underlying price data
-        /// </summary>
-        public BaseData Underlying
-        {
-            get { return _underlying; }
-        }
-
-        internal BaseData _underlying;
-
-        /// <summary>
-        /// True if the universe is dynamic and filter needs to be reapplied
-        /// </summary>
-        public bool IsDynamic
-        {
-            get
-            {
-                return _isDynamic;
-            }
-        }
-
-        internal bool _isDynamic;
-
         /// <summary>
         /// Constructs FutureFilterUniverse
         /// </summary>
```

Comment:
I think it would be nice if this constructor should be calling the base one and not setting those fields here?